resources/api/reservation.py

```diff
@@ -178,6 +178,9 @@ def __init__(self, *args, **kwargs):
                 cache=cache
             )
 
+            if invoice_requested:
+                supported = set(supported) | invoicing_specific_fields
```

Comment:
Maybe missing some context: where else is `supported` used here?

Reply:
Why these changes are needed:

Customer wants to allow payment by invoice for a resource but they don't want to have the invoicing related fields required when the user does not request payment by invoice. Therefore, these fields should not be in the metadata set but they still need to be saved to the reservation when payment by invoice is requested. That's why they need to be in the supported fields here.